mechafil/data.py

```diff
@@ -4,198 +4,108 @@
 import datetime
 from typing import Tuple
 
-EXBI = 2**60
-PIB = 2**50
-
+from . import data_spacescope, data_starboard
+DEFAULT_DATA_BACKEND = 'spacescope'
 
 def get_historical_network_stats(
-    start_date: datetime.date, current_date: datetime.date, end_date: datetime.date
+    start_date: datetime.date, 
+    current_date: datetime.date, 
+    end_date: datetime.date,
+    data_backend: str = DEFAULT_DATA_BACKEND
 ) -> pd.DataFrame:
-    power_df = query_starboard_power_stats(start_date, current_date)
-    onboards_df = query_starboard_daily_power_onboarded(start_date, current_date)
-    stats_df = query_starboard_supply_stats(start_date, current_date)
-    stats_df = stats_df.merge(power_df, on="date", how="inner").merge(
-        onboards_df, on="date", how="inner"
-    )
-    rb_renewal_rate, day_renewed_qa_power = get_day_renewed_power_stats(
-        start_date, current_date, end_date
-    )
-    stats_df["rb_renewal_rate"] = rb_renewal_rate
-    stats_df["day_renewed_qa_power_pib"] = day_renewed_qa_power
-    return stats_df
+    if data_backend.lower() == 'spacescope':
+        return data_spacescope.get_historical_network_stats(start_date, current_date, end_date)
+    elif data_backend.lower() == 'starboard':
+        return data_starboard.get_historical_network_stats(start_date, current_date, end_date)
+    else:
+        raise ValueError("Data Backend: %s not supported!" % (data_backend,))
 
 
 def get_sector_expiration_stats(
     start_date: datetime.date,
     current_date: datetime.date,
     end_date: datetime.date,
+    data_backend: str = DEFAULT_DATA_BACKEND
```

Comment:
I think it is a good idea to give the user the option of the backend. 

Having this in mind, we should also give the option when people just want to run the simulation directly through `run_simple_sim` from the `sim.py` module. Thus, I think we should make two additional changes:
1. Add `data_backend: str = DEFAULT_DATA_BACKEND` to the  `run_simple_sim` function and all the other functions that read data (from `minting.py` and `vesting.py`).
2. Add a note in the readme explaining this parameter and the difference between using the two backends

Note that the readme would also be a good place to explain how to setup a key in spacescope.

Reply:
I largely left starboard in there for backwards compatibility, but I am not sure that it is worth the effort to do all of this.  Starboard is supposed to have been out of commission starting 2023-01, but it looks like they are still keeping it on for another month or maybe two.  Beyond that, there will only be one backend and that will be spacescope.  My opinion is that considering this, we can leave things as they are b/c I suspect in 2 months, we'll just go into the code and delete starboard altogether.  

But, let me know your thoughts - its not difficult to add in but IMO its not worth the effort.

Reply:
If the user won't have a real choice (since Starboard will stop working), then I agree that there is not need to the extra work.